pkg/gcptarget/gcptarget.go

```diff
@@ -0,0 +1,224 @@
+package gcptarget
+
+import (
+	"bytes"
+	"encoding/json"
+	"regexp"
+	"strings"
+	"text/template"
+
+	"github.com/forseti-security/config-validator/pkg/api/validator"
+	asset2 "github.com/forseti-security/config-validator/pkg/asset"
+	"github.com/gogo/protobuf/jsonpb"
+	"github.com/open-policy-agent/frameworks/constraint/pkg/client"
+	"github.com/open-policy-agent/frameworks/constraint/pkg/types"
+	"github.com/pkg/errors"
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+// Name is the target name for GCPTarget
+const Name = "validation.gcp.forsetisecurity.org"
+
+// GCPTarget is the constraint framework target for FCV
+type GCPTarget struct {
+}
+
+var _ client.TargetHandler = &GCPTarget{}
+
+// New returns a new GCPTarget
+func New() *GCPTarget {
+	return &GCPTarget{}
+}
+
+// MatchSchema implements client.MatchSchemaProvider
+func (g *GCPTarget) MatchSchema() apiextensions.JSONSchemaProps {
+	return apiextensions.JSONSchemaProps{
+		Properties: map[string]apiextensions.JSONSchemaProps{
+			"target": {
+				Type: "array",
+				Items: &apiextensions.JSONSchemaPropsOrArray{
+					Schema: &apiextensions.JSONSchemaProps{
+						Type: "string",
+					},
+				},
+			},
+			"exclude": {
+				Type: "array",
+				Items: &apiextensions.JSONSchemaPropsOrArray{
+					Schema: &apiextensions.JSONSchemaProps{
+						Type: "string",
+					},
+				},
+			},
+		},
+	}
+}
+
+// GetName implements client.TargetHandler
+func (g *GCPTarget) GetName() string {
+	return Name
+}
+
+// Library implements client.TargetHandler
+func (g *GCPTarget) Library() *template.Template {
+	return libraryTemplate
+}
+
+// ProcessData implements client.TargetHandler
+func (g *GCPTarget) ProcessData(obj interface{}) (bool, string, interface{}, error) {
+	return false, "", nil, errors.Errorf("Storing data for referential constraint eval is not supported at this time.")
+}
+
+// HandleReview implements client.TargetHandler
+func (g *GCPTarget) HandleReview(obj interface{}) (bool, interface{}, error) {
+	switch asset := obj.(type) {
+	case *validator.Asset:
+		return g.handleAsset(asset)
+	case map[string]interface{}:
+		if _, found, err := unstructured.NestedString(asset, "name"); !found || err != nil {
+			return false, nil, err
+		}
+		if _, found, err := unstructured.NestedString(asset, "asset_type"); !found || err != nil {
+			return false, nil, err
+		}
+		if _, found, err := unstructured.NestedString(asset, "ancestry_path"); !found || err != nil {
+			return false, nil, err
+		}
+		_, foundResource, err := unstructured.NestedMap(asset, "resource")
+		if err != nil {
+			return false, nil, err
+		}
+		_, foundIam, err := unstructured.NestedMap(asset, "iam_policy")
+		if err != nil {
+			return false, nil, err
+		}
+		if !foundIam && !foundResource {
+			return false, nil, nil
+		}
+		if foundIam && foundResource {
+			return false, nil, errors.Errorf("malformed asset has iam_policy and resource: %v", asset)
```

Comment:
I'm not sure if this actually an error? I don't see any intrinsic issue with one asset having both.